service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -66,43 +69,43 @@
     private final ParticipantMapper participantMapper;
     private final RandomIdGeneratorService randomIdGeneratorService;
 
-    public String mapObservationToCompoundStatement(Observation observationAssociatedWithSpecimen, List<Observation> observations) {
+    public String mapObservationToCompoundStatement(Observation observationAssociatedWithSpecimen) {
```

Comment:
With the amount of parameters being passed around for this now I'd be thinking about using an inner class to hold state. What do you think?

Reply:
It adds a layer of abstraction (which may or may not be a good thing), and makes the processes within that layer a lot simpler - syntactically for sure, as the method signatures become cleaner, and imo in terms of readability too. You end up with a thing whose responsibility, and entire lifespan, is just processing a single mapping. The cost is in understanding that that is what is happening which could be confusing for readers who aren't familiar with the strategy, and... some extra indentation?

Both will work so I'm certainly not going to insist, but there's my suggestion! 😀